integrations/node/workload-executor.js

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env node
+
+const MongoClient = require('../../node-mongodb-native').MongoClient;
+const fs = require('fs');
+const assert = require('assert');
+const omit = require('lodash.omit');
+
+let interrupted = false;
+process.on('SIGINT', () => (interrupted = true));
+
+const executors = {
+  find: (collection, args) =>
+    collection.find(args.filter || {}, omit(args, ['filter'])).toArray(),
+
+  insertOne: (collection, args) =>
+    collection.insertOne(args.document, { forceServerObjectId: true }),
+
+  updateOne: (collection, args) =>
+    collection.updateOne(args.filter, args.update)
+};
+
+async function main(uri, spec) {
+  const runOperations = makeRunOperations(spec);
+
+  const client = new MongoClient(uri);
+  await client.connect();
+
+  const results = { numSuccesses: 0, numFailures: 0, numErrors: 0 };
+
+  while (!interrupted) {
+    try {
+      await runOperations(client, results);
+    } catch (err) {
+      ++results.numErrors;
+    }
+  }
+
+  fs.writeFileSync('results.json', JSON.stringify(results));
+  await client.close();
+
+  process.exit(0);
+}
+
+function makeRunOperations(spec) {
+  const operations = spec.operations;
+  return function(client, results) {
+    if (!Array.isArray(operations) || operations.length === 0) {
+      return new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
can we use a more descriptive constant for this magic 1000 number?

Reply:
That timeout actually carried over from the original sample code you sent me, so I wasn't 100% sure what it was for, but it looked like it was there to throttle empty requests. I've renamed it to `NOOP_DELAY`